tables/Tables/Table.cc

```diff
@@ -527,19 +508,19 @@ BaseTable* Table::makeBaseTable (const String& name, const String& type,
     ios >> format;
     ios >> tp;
     if (tp == "PlainTable") {
-	baseTabPtr = new PlainTable (ios, version, name, type, nrrow,
-				     tableOption, lockOptions, tsmOpt,
-                                     addToCache, locknr);
+      baseTabPtr.reset (new PlainTable (ios, version, name, type, nrrow,
+                                        tableOption, lockOptions, tsmOpt,
+                                        addToCache, locknr));
```

Comment:
It is generally advised to always use `ptr = std::make_shared()` instead of `ptr.reset(new ...)`, because the second one costs one more allocation. This would then become `baseTabPtr = std::make_shared<PlainTable>(ios, version, ...)`. This also holds for the other cases below.

Reply:
std::make_shared can only be used for concrete classes, not for abstract ones. Table is using std::shared_ptr<BaseTable>, thus for an abstract class. I realize that a std::shared_ptr<PlainTable> can be casted to a std::shared_ptr<BaseTable>, but for the few times a PlainTable object is allocated, this sounds like overkill to me.